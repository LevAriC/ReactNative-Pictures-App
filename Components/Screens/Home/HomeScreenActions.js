import { TOGGLE_VIEW, SEARCH_PICS } from './HomeScreenActionTypes'

const toggleView = (view) => ({
    type: TOGGLE_VIEW,
    data: { view }
})
const handleToggleView = (view) => async dispatch => {
    dispatch(toggleView(view))
}

const updatePicsList = (picList) => ({
    type: SEARCH_PICS,
    data: { picList }
})
const handlePicsSearch = (searchQuery) => async dispatch => {
    const pixabayKey = '12263176-901abf3c360dd1fbe2da948f7'
    const url = `https://pixabay.com/api/?key=${pixabayKey}&q=${searchQuery}&image_type=photo&per_page=199`
    const options = { method: 'GET', headers: { 'Content-Type': 'application/json' } }
    const request = new Request(url, options)
    console.log(request)

    await fetch(request)
    .then(response => response.json())
    .then(async data => dispatch(updatePicsList(data.hits)))
    .catch(err => console.log(err))
}

export default {
    handlePicsSearch,
    handleToggleView,
    updatePicsList,
    toggleView
}