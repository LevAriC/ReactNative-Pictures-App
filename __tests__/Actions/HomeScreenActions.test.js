import HomeScreenActions from '../../Components/Screens/Home/HomeScreenActions'

it("Testing Action's type", () => {
    expect(HomeScreenActions.updatePicsList().type).toMatchSnapshot();
    expect(HomeScreenActions.toggleView().type).toMatchSnapshot();
});