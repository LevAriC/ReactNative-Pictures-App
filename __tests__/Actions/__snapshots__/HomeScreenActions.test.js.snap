// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Action's type 1`] = `"SEARCH_PICS"`;

exports[`Testing Action's type 2`] = `"TOGGLE_VIEW"`;
